workflow/scripts/savetoh5.jl

```diff
@@ -0,0 +1,176 @@
+
+Convert the centroid coordinates from row and column to latitude and longitude. Also drop the columns specified in `colstodrop`.
+"""
+function convertcentroid!(propdf, latlondata, colstodrop)
+    latitude, longitude = [
+        [
+            latlondata[c][Int(round(_x)), Int(round(_y))]]
+        for (_x, _y) in zip(propdf.row_centroid, propdf.col_centroid)
+        for c in ["latitude", "longitude"]
+    ]
+    propdf.latitude = latitude
+    propdf.longitude = longitude
+    dropcols!(propdf, colstodrop)
+end
+
+function dropcols!(df, colstodrop)
+    select!(df, Not(colstodrop))
+    return nothing
+end
+
+function converttounits!(propdf, latlondata, colstodrop)
+    if nrow(propdf) == 0
+        dropcols!(propdf, colstodrop)
+        return nothing
+    end
+    convertcentroid!(propdf, latlondata, colstodrop)
+    x = latlondata["X"]
+    dx = abs(x[2] - x[1])
+    convertarea(area) = area * dx^2 / 1e6
+    convertlength(length) = length * dx / 1e3
+    propdf.area .= convertarea(propdf.area)
+    propdf.convex_area .= convertarea(propdf.convex_area)
+    propdf.minor_axis_length .= convertlength(propdf.minor_axis_length)
+    propdf.major_axis_length .= convertlength(propdf.major_axis_length)
+    propdf.perimeter .= convertlength(propdf.perimeter)
+    return nothing
+end
+
+@pyinclude(joinpath(@__DIR__, "latlon.py"))
+
+getlatlon = py"getlatlon"
+
+"""
+    makeh5file(pathtosampleimg, resdir)
+
+Package the results of the IceFloeTracker pipeline in `resdir` into individual HDF5 files in `resdir/hdf5-files`. 
+
+This function expects the following files to be present in `resdir`: `filenames.jls`, `passtimes.jls`, `segmented_floes.jls`, and `floe_props.jls`. These files are generated by the `IceFloeTracker` pipeline.
+
+# Arguments:
+
+  * `pathtosampleimg`: Path to a sample image. This is used to extract the coordinate reference system (CRS) and the latitude and longitude coordinates of the image pixels.
+  * `resdir`: Path to the directory containing the results of the IceFloeTracker pipeline.
+
+# File structure
+Each HDF5 file has the following structure:
+
+```
+🗂️ HDF5.File: (read-only) YYYYMMDD.sat.labeled_image.250m.h5
+├─ 🏷️ contact
+├─ 🏷️ crs
+├─ 🏷️ fname_reflectance
+├─ 🏷️ fname_truecolor
+├─ 🏷️ iftversion
+├─ 🏷️ reference
+├─ 📂 floe_properties
+│  ├─ 🏷️ Description of labeled_image
+│  ├─ 🏷️ Description of properties
+│  ├─ 🔢 colunm_names
+│  ├─ 🔢 labeled_image
+│  └─ 🔢 properties
+└─ 📂 index
+   ├─ 🔢 latitude
+   ├─ 🔢 longitude
+   ├─ 🔢 time
+   ├─ 🔢 x
+   └─ 🔢 y
+```
+# The `floe_properties` and `index` group
+
+The `floe_properties` group contains a floe properties matrix `properties` for `labeled_image` and assciated `colunm_names`.
+
+The `index` group contains the latitude and longitude coordinates (see the description of properties within the file for an account of the units of each property), and satellite pass time `time` in Unix time that captured the source image.
```

Comment:
Needs the polar stereographic index too. Perhaps change the text to the following:

The 'index' group contains the spatial coordinates in the source image coordinate reference system (default NSIDC polar stereographic, meters) and geographic coordinates (latitude and longitude, decimal degrees). Estimated satellite overpass time 'time' is provided in Unix time (seconds since 1970-01-01 00:00 UTC).